src/plugins/action_generate_review_assignment_records/src/generateReviewAssignments.ts

```diff
@@ -61,19 +65,44 @@ module.exports['generateReviewAssignments'] = async function (input: any, DBConn
           isLastLevel: nextReviewLevel === numReviewLevels,
         }
     })
+    // console.log('ReviewAssignments', reviewAssignments)
+
+    // Save review_assignment records to database
+    const reviewAssignmentIds = await db.addReviewAssignments(Object.values(reviewAssignments))
 
-    const reviewAssignmentIds = await DBConnect.addReviewAssignments(
-      Object.values(reviewAssignments)
+    // Generate review_assignment_assigner_joins
+    // For now we assume that assigners have no Section restrictions
+    console.log('Generating review_assignment_assigner_join records...')
+    const availableAssigners = await db.getPersonnelForApplicationStageLevel(
+      templateId,
+      stageNumber,
+      nextReviewLevel,
+      'Assign'
+    )
+    const reviewAssignmentAssignerJoins = []
+    for (const reviewAssignmentId of reviewAssignmentIds) {
+      for (const assigner of availableAssigners) {
+        reviewAssignmentAssignerJoins.push({
+          assignerId: assigner.userId,
+          orgId: assigner.orgId,
+          reviewAssignmentId,
```

Comment:
still not sure how this would work if there is an error (instead of id) as reviewAssignmentId.

I forgot what happens if an action thrown uncaught error (if it's gracefully handles further up or not)

Reply:
The whole Action is in a `try{ }` block, and the `catch` stores the error message in its output log, which gets written to the action_queue along with Success/Fail etc.

So the Action should throw the error on the first error and never reach here if there was a problem with generating reviewAssignmentIds.

Reply:
```
 } catch (error) {
    console.log(error.message)
    return {
      status: 'Fail',
      error_log: 'Problem creating review_assignment records: ' + error.message,
    }
  }
```
Every Action follows this same template, FYI.